libs/common-hooks/src/useTheme.ts

```diff
@@ -1,7 +1,26 @@
-import { useContext } from 'react'
+'use client'
 
-import { ThemeContext } from 'styled-components/macro'
+import styled from 'styled-components/macro'
 
-export function useTheme() {
-  return useContext(ThemeContext)
+import type { DefaultTheme } from 'styled-components/macro'
+
+import '@cowprotocol/types'
+
+// Re-export the theme hook with proper typing
+export function useTheme(): DefaultTheme {
+  // SSR-safe check
+  const isServer = typeof window === 'undefined'
+
+  if (isServer) {
```

Comment:
Why can't we use `styled-components/macro` on server side?

Reply:
Apparently Next.js server components run before client-side build tools kick in, so Babel macros (like styled-components/macro) aren't available there. At least I was getting build errors exclusively for `cow-fi` app when trying to build it. This was what made all builds (for all our apps) consistently work.